gio/src/list_store.rs

```diff
@@ -75,3 +76,23 @@ unsafe extern "C" fn compare_func_trampoline(
 
     (*func)(&a, &b).into_glib()
 }
+
+impl<A: IsA<glib::Object>> std::iter::Extend<A> for ListStore {
```

Comment:
Maybe also for `std::iter::Extend<&A>`?

Reply:
It would be possible to do this without cloning and without allocating a second `Vec`, if something like this was done instead:

```rust
impl<'a, A: IsA<glib::Object>> std::iter::Extend<&'a A> for ListStore
```

Reply:
> It would be possible to do this without cloning and without allocating a second Vec

How? If you insert item by item while iterating then this causes a lot of overhead (signals!). Unfortunately this can't be efficiently re-implemented outside of `gliststore.c`.

Reply:
I can't implement both `Extend<A>` and `Extend<&'a A>`.
```
conflicting implementation for `auto::list_store::ListStore`
note: downstream crates may implement trait `glib::IsA<glib::Object>` for type `&'a _`
```

Also, `splice` needs a slice, so as sdroege said, I have to allocate a `Vec` somewhere. The `ListStore` implementation is in C and it takes a slice.